code/lightna/magento-os-frontend/js/common/ShoppingCart.js

```diff
@@ -5,117 +5,193 @@ import { $, $$ } from '../lib/utils';
-
-        request.post('/checkout/sidebar/removeItem/', data, {
-            onSuccess: () => {
-                itemToRemove.classList.add('fade-out');
-                setTimeout(() => {
-                    this.updateMinicart.bind(this)();
-                }, 200);
-            },
-        });
+
+        request
+          .post(this.addToCartUrl, data, {
+              onSuccess: this.addProductHandler.bind(this),
+              showMessage: false,
+          })
+          .then(() => {
+              this.toggleAnimation(actionTrigger, false);
+          });
     }
 
-    updateMinicart() {
-        const updateMinicartUrl = document.location.pathname + '?block=.body.minicart-content';
+    removeProduct(itemId) {
+        const data = {
+            item_id: itemId,
+        };
+        const itemToRemove = $(
+          `[data-item-id="${itemId}"]`,
+          $(this.selectors.shoppingCart),
+        ).closest('li');
 
         request
-            .post(
-                updateMinicartUrl,
-                {},
-                {
-                    onSuccess: this.updateMinicartHtml.bind(this),
-                },
-            )
-            .then(() => {
-                this.updateMagentoCartSectionId();
-            });
+          .post(this.removeFromCartUrl, data, {
+              onSuccess: this.removeProductHandler.bind(this, itemToRemove),
+          })
+          .then();
     }
 
-    updateMinicartHtml(html) {
-        $('.cjs-minicart-content').outerHTML = html;
-        this.addCartEventListeners();
-        this.updateCounter();
+    addToCartHandler(trigger) {
+        const addToCartComponent = trigger.closest(
+          this.selectors.addToCartComponent,
+        );
+        this.addProduct(addToCartComponent, trigger);
+        this.toggleAnimation(trigger, true);
     }
 
-    updateMagentoCartSectionId() {
-        let sids = Cookie.get('section_data_ids');
-        sids = sids ? JSON.parse(decodeURIComponent(sids)) : {};
-        sids.cart = sids.cart ? sids.cart + 1000 : 1;
+    addProductHandler(response) {
+        // todo: add type of message to response
+        const temp = document.createElement('div');
+        temp.innerHTML = response.messagesHtml;
+        if (temp.querySelector('.message').classList.contains('success')) {
+            this.loaded ? this.update.bind(this)() : this.load.bind(this)();
+        } else {
+            this.showResponseMessage(response);
+        }
+    }
 
-        Cookie.set('section_data_ids', encodeURIComponent(JSON.stringify(sids)));
+    removeProductHandler(item) {
+        item.classList.add(this.classes.fade);
+        setTimeout(() => {
+            this.update.bind(this)();
+        }, this.minActionDuration);
     }
 
     updateCounter() {
-        $('.cjs-minicart-counter').innerText = $('.minicart').dataset.totalQty;
+        $(this.selectors.shoppingCartCounter).innerText = $(
+          this.selectors.shoppingCart,
+        ).dataset.totalQty;
     }
```

Comment:
- counter is not the part of the block we update
- was thinking to place all the overlay in the bottom of the page

Reply:
counter is not the part of the block we update - but could be, and it will simplify
was thinking to place all the overlay in the bottom of the page - if it's empty on page load then there is no differnce as for me